ui/app/components/app/modals/customize-nonce/customize-nonce.component.js

```diff
@@ -0,0 +1,128 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Modal from '../../modal';
+import TextField from '../../../ui/text-field';
+
+export default class CustomizeNonce extends PureComponent {
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    customNonceValue: PropTypes.string,
+    nextNonce: PropTypes.number,
+    updateCustomNonce: PropTypes.func,
+    getNextNonce: PropTypes.func,
+  };
+
+  static contextTypes = {
+    t: PropTypes.func,
+  };
+
+  state = {
+    customNonceValue: '',
+  };
+
+  onChange = (e) => {
+    this.setState({
+      customNonceValue: e.target.value,
+    });
+  };
+
+  onSubmit = () => {
+    const { customNonceValue } = this.props;
+    if (this.state.customNonceValue === '') {
+      this.props.updateCustomNonce(customNonceValue);
+    } else {
+      this.props.updateCustomNonce(this.state.customNonceValue);
+    }
+    this.props.getNextNonce();
+    this.props.hideModal();
+  };
+
+  onReset = () => {
+    const { nextNonce } = this.props;
+    this.setState({
+      customNonceValue: nextNonce,
+    });
+    document.getElementById('custom-nonce-id').value = nextNonce;
+  };
+
+  renderModalContent() {
+    const { t } = this.context;
+    const { hideModal, nextNonce, customNonceValue } = this.props;
+    return (
+      <div className="customize-nonce-modal">
+        <div className="customize-nonce-modal__main-header">
+          <div className="customize-nonce-modal__title">
```

Comment:
[REQ] We should strive to create as semantic of HTML as possible within our time and knowledge constraints. A good rule of thumb is to use as few divs as possible to express your design. You can't avoid using divs, but if you first ask "what else could this be" it usually helps. In this case when you have a div that only is getting typography styles and contains text, using a heading tag, paragraph tag or even a span tag would be better than div. You could also express this without writing any additional custom css by using the `Typography` component.

```jsx
<Typography variant={TYPOGRAPHY.H4} fontWeight={FONT_WEIGHT.BOLD}>
 {t('editNonceField')}
</Typography>
```

this will render with a `h4` tag (matching the variant level), but you can also render whatever HTML tag you'd prefer:

```jsx
<Typography variant={TYPOGRAPHY.H4} fontWeight={FONT_WEIGHT.BOLD} tag={'h2'}>
 {t('editNonceField')}
</Typography>
```

In general, you'd want only one h1 per page, and logically increment heading levels based on any predecessor headings -- but we haven't implemented any strictness in that regard yet. h4 is probably fine here.